table/data_block_index_format.cc

```diff
@@ -0,0 +1,43 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#include "data_block_index_format.h"
+
+#include "rocksdb/table.h"
+
+namespace rocksdb {
+
+const int kDataBlockIndexTypeBitShift = 31;
+
+// 0x7FFFFFFF
+const uint32_t kMaxNumRestarts =  (1u << kDataBlockIndexTypeBitShift) - 1u;
+
+// 0x7FFFFFFF
+const uint32_t kNumRestartsMask = (1u << kDataBlockIndexTypeBitShift) - 1u;
+
+uint32_t PackIndexTypeAndNumRestarts(
+    BlockBasedTableOptions::DataBlockIndexType index_type,
+    uint32_t num_restarts) {
+  assert(num_restarts <= kMaxNumRestarts);
+  return num_restarts |= index_type << kDataBlockIndexTypeBitShift;
+}
+
+
+void UnPackIndexTypeAndNumRestarts(
```

Comment:
nit: this should be "Unpack" with a lower-case p.

Reply:
I copy this format from
https://github.com/facebook/rocksdb/blob/5.15.fb/db/dbformat.cc#L67. I guess they want to emphsize it is a reverse operation of Pack?